internal/contour/virtualhost.go

```diff
@@ -28,6 +29,28 @@ type VirtualHostCache struct {
 	Cond
 }
 
+func getRequestTimeout(annotations map[string]string) *time.Duration {
+	timeoutStr, ok := annotations["contour.heptio.com/request-timeout"]
```

Comment:
you can return 0 here, that's a valid time.Duration.

Reply:
The problem is `0` as a value and being passed in to envoy has a different semantic (don't ever timeout) than `not specified` which means "use the default timeout". I'm happy enough to say contour always does the "no timeout" by default, but that is changing envoy default behavior / making it special in the case of "used via envoy". Also changes the behavior in existing clusters.